synapse/models/infotech.py

```diff
@@ -540,6 +627,180 @@ def getModelDefs(self):
+                        'ro': True,
+                        'doc': 'The "sw_edition" field from the CPE 2.3 string.'}),
+                    ('target_sw', ('str', {'lower': True, 'strip': True}), {
+                        'ro': True,
+                        'doc': 'The "target_sw" field from the CPE 2.3 string.'}),
+                    ('target_hw', ('str', {'lower': True, 'strip': True}), {
+                        'ro': True,
+                        'doc': 'The "target_hw" field from the CPE 2.3 string.'}),
+                    ('other', ('str', {'lower': True, 'strip': True}), {
+                        'ro': True,
+                        'doc': 'The "other" field from the CPE 2.3 string.'}),
+                )),
+                ('it:sec:cwe', {}, (
+                    ('name', ('str', {}), {
+                        'doc': 'The CWE description field.',
+                        'ex': 'Buffer Copy without Checking Size of Input (Classic Buffer Overflow)',
+                    }),
+                    ('desc', ('str', {}), {
+                        'doc': 'The CWE description field.',
+                        'disp': {'hint': 'text'},
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'A URL linking this CWE to a full description.',
+                    }),
+                    ('parents', ('array', {'type': 'it:sec:cwe', 'uniq': True, 'split': ','}), {
+                        'doc': 'An array of ChildOf CWE Relationships.'
+                    }),
+                )),
+                ('it:mitre:attack:group', {}, (
+                    ('org', ('ou:org', {}), {
+                        'doc': 'Used to map an ATT&CK group to a synapse ou:org.',
+                    }),
+                    ('name', ('ou:name', {}), {
+                        'doc': 'The primary name for the ATT&CK group.',
+                    }),
+                    ('names', ('array', {'type': 'ou:name', 'uniq': True}), {
+                        'doc': 'An array of alternate names for the ATT&CK group.',
+                    }),
+                    ('desc', ('str', {}), {
+                        'doc': 'A description of the ATT&CK group.',
+                        'disp': {'hint': 'text'},
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The URL that documents the ATT&CK group.',
+                    }),
+                    ('tag', ('syn:tag', {}), {
+                        'doc': 'The synapse tag used to annotate nodes included in this ATT&CK group ID.',
+                        'ex': 'cno.mitre.g0100',
+                    }),
+                    ('references', ('array', {'type': 'inet:url'}), {
+                        'doc': 'An array of URLs that document the ATT&CK group.',
+                    }),
+                    ('techniques', ('array', {'type': 'it:mitre:attack:technique', 'uniq': True, 'split': ','}), {
```

Comment:
Let's discuss pros/cons of encoding / cross-referencing :techniques / :software / :groups vs. leaving this off (too much to type / talk about).

Reply:
Def want to make it possible to record these relationships, but i think we'll probably roll out a storm package to automatically sync them mitre as the authoritative source.  Def dont want analysts having to manually fill these out. :D